src/mds/Locker.cc

```diff
@@ -3215,6 +3215,8 @@ void Locker::_update_cap_fields(CInode *in, int dirty, const MClientCaps::const_
       dout(7) << "  mtime " << pi->mtime << " -> " << mtime
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
+      if (mtime > pi->rstat.rctime)
+	pi->rstat.rctime = mtime;
```

Comment:
Shouldn't this change not be necessary if the client updates the ctime correctly now?

Reply:
this is for buggy client